lib/chef/resource/windows_certificate.rb

```diff
@@ -277,7 +277,9 @@ def acl_script(hash)
         def convert_pem(ext)
           out = case ext
                 when ".crt", ".cer", ".der"
-                  powershell_out("openssl x509 -text -inform DER -in #{new_resource.source} -outform PEM")
+                  command = "openssl x509 -text -in #{new_resource.source} -outform PEM"
+                  command += " -inform DER" if binary_cert?
+                  powershell_out(command)
```

Comment:
If we can't do this with pure ruby we should use powershell_out! here so it fails if this fails

Reply:
Actually, I deliberately removed `powershell_out!`. In case of any error, it shows all the commands and details which was not looking user friendly. Pretty printing of an error was the only reason that I used `powershell_out`. I can revert back the code in case of any issues. Please do let me know.

Here is the case:

**powershell_out!("Some invalid command")**
```
*** Mixlib::ShellOut::ShellCommandFailed Exception: Expected process to exit with [0], but received '1'
---- Begin output of powershell.exe -NoLogo -NonInteractive -NoProfile -ExecutionPolicy Unrestricted -InputFormat None -Command "Some invalid command" ----
STDOUT:
STDERR: Some : The term 'Some' is not recognized as the name of a cmdlet, function, script file, or operable program. Check
the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ Some invalid command
+ ~~~~
    + CategoryInfo          : ObjectNotFound: (Some:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
---- End output of powershell.exe -NoLogo -NonInteractive -NoProfile -ExecutionPolicy Unrestricted -InputFormat None -Command "Some invalid command" ----
Ran powershell.exe -NoLogo -NonInteractive -NoProfile -ExecutionPolicy Unrestricted -InputFormat None -Command "Some invalid command" returned 1

```


**powershell_out("Some invalid command").stderr**

```
"Some : The term 'Some' is not recognized as the name of a cmdlet, function, script file, or operable program. Check \r\nthe spelling of the name, or if a path was included, verify that the path is correct and try again.\r\nAt line:1 char:1\r\n+ Some invalid command\r\n+ ~~~~\r\n    + CategoryInfo          : ObjectNotFound: (Some:String) [], CommandNotFoundException\r\n    + FullyQualifiedErrorId : CommandNotFoundException\r\n \r\n"
```